### PR TITLE
refactor: migrate giphy to react (#ACC-218)

### PR DESCRIPTION
**Pull request by przemvs**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-218" title="ACC-218" target="_blank">TaskACC-218</a>  [Web] Migrate giphy component to React
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: src/script/components/Giphy/index.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  };
+
+  const onGridClick = async () => getGifs();
+
+  const onBackClick = () => {
+    if (currentGif) {
+      setGifs([currentGif]);
+      setSelectedGif(currentGif);
+    }
+    setGiphyState(GiphyState.RESULT);
```
Comment (PatrykBuniX) [inline src/script/components/Giphy/index.tsx]: shouldn't this line be also inside the if statement?

File: src/script/components/Giphy/index.tsx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+      }
+    } catch (error) {
+      console.warn(error);
+      setGiphyState(GiphyState.ERROR);
+
+      logger.warn(error);
+    }
+  };
+
+  const onGridClick = async () => getGifs();
```
Comment (PatrykBuniX) [inline src/script/components/Giphy/index.tsx]: There is no need to wrap `getGifs` function, we can just use it: `onClick={getGifs}`

File: src/script/components/Giphy/index.tsx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+  return (
+    <div
+      role="dialog"
+      aria-labelledby="giphy-name"
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+      tabIndex={0}
+    >
+      <div className="modal-content">
+        <div className="giphy-modal-header modal-header">
+          {isResultState && <button className="button-icon icon-grid" onClick={onGridClick} />}
```
Comment (PatrykBuniX) [inline src/script/components/Giphy/index.tsx]: add accessible `aria-label`

File: src/script/components/Giphy/index.tsx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+          {(isResultsState || isNoSearchResultState) && (
+            <button className="button-icon icon-back" onClick={onBackClick} data-uie-name="do-close" />
```
Comment (PatrykBuniX) [inline src/script/components/Giphy/index.tsx]: add accessible `aria-label`

File: src/script/components/Giphy/index.tsx [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+  const onCloseClick = () => {
+    giphyModal.hide();
+    giphyRepository.resetOffset();
+  };
+
+  const showGiphy = async (query: string) => {
+    setQuery(query);
+    setGiphyState(GiphyState.DEFAULT);
+    await getGifs(true);
```
Comment (PatrykBuniX) [inline src/script/components/Giphy/index.tsx]: ⚠️ breaking change: `getGifs` will use currently closed value of `query` (closure), not the one that is being set two lines above (line `128`). We can: 
1. wait for the state to update (`useEffect` with `[query]` deps) and then call `getGifs(true)`, or
2. add query param to `getGifs` function and call it with updated `query` value: `getGifs(query, true)`

I prefer 2nd option 😄

**Review by PatrykBuniX — COMMENTED** [review]
Did you test it locally?

File: src/script/components/Giphy/index.tsx (hunk 5 above)
Comment (przemvs) [reply src/script/components/Giphy/index.tsx]: Ahh.. you're sure! :D I miss it in refactoring :D

File: src/script/components/Giphy/index.tsx (hunk 2 above)
Comment (przemvs) [reply src/script/components/Giphy/index.tsx]: Now need, because i'm passing query to getGifs. :)

Comment (PatrykBuniX) [reply src/script/components/Giphy/index.tsx]: 👌

**Review by thisisamir98 — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]
looks good

File: src/script/components/Giphy/index.tsx [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+        setGiphyState(GiphyState.RESULT);
+      } else {
+        setGifs(fetchedGifs);
+        setGiphyState(GiphyState.RESULTS);
+      }
+    } catch (error) {
+      console.warn(error);
+      setGiphyState(GiphyState.ERROR);
+
+      logger.warn(error);
```
Comment (tlebon) [inline src/script/components/Giphy/index.tsx]: do we want logger and console warn errors?

File: src/script/components/Giphy/index.tsx [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+
+  useEffect(() => {
+    amplify.subscribe(WebAppEvents.EXTENSIONS.GIPHY.SHOW, showGiphy);
+  }, []);
+
+  return (
+    <div
+      role="dialog"
+      aria-labelledby="giphy-name"
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
```
Comment (tlebon) [inline src/script/components/Giphy/index.tsx]: im wondering about this in the context of accessibility

File: src/script/components/Giphy/index.tsx [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+  const [gifs, setGifs] = useState<Gif[]>([]);
+  const [selectedGif, setSelectedGif] = useState<Gif | null>(null);
+  const [currentGif, setCurrentGif] = useState<Gif | null>(null);
+
+  const [giphyState, setGiphyState] = useState<GiphyState>(defaultGiphyState);
+  const isErrorState = giphyState === GiphyState.ERROR;
+  const isLoadingState = giphyState === GiphyState.LOADING;
+  const isResultState = giphyState === GiphyState.RESULT;
+  const isResultsState = giphyState === GiphyState.RESULTS;
+  const areResultsState = [GiphyState.RESULT, GiphyState.RESULTS].includes(giphyState);
```
Comment (tlebon) [inline src/script/components/Giphy/index.tsx]: these vars are named too closely for me.

File: src/script/components/Giphy/index.tsx (hunk 6 above)
Comment (przemvs) [reply src/script/components/Giphy/index.tsx]: I think we don't need. So i'll remove logger from this component.

File: src/script/components/Giphy/index.tsx (hunk 7 above)
Comment (przemvs) [reply src/script/components/Giphy/index.tsx]: Removed

File: src/script/components/Giphy/index.tsx (hunk 8 above)
Comment (przemvs) [reply src/script/components/Giphy/index.tsx]: done

**Review by V-Gira — APPROVED** [review]

**Review by tlebon — APPROVED** [review]
Lgtm

**Pull request merged**